src/proof.rs

```diff
@@ -21,29 +25,53 @@ pub fn blind_bid_proof(
     value: JubJubScalar,
     blinder: JubJubScalar,
 ) -> Result<()> {
+    // Generate constant witness values for 0.
+    let zero = composer.add_witness_to_circuit_description(BlsScalar::zero());
     // Get the corresponding `StorageBid` value that for the `Bid`
     // which is effectively the value of the proven leaf.
     let storage_bid = StorageBid::from(bid);
     let encoded_bid: StorageScalar = storage_bid.into();
     let proven_leaf = composer.add_input(encoded_bid.into());
+    // Allocate bid-needed inputs
+    let latest_consensus_step =
+        AllocatedScalar::allocate(composer, bid.latest_consensus_step);
```

Comment:
It is unclear to me how we reconstruct the `latest_consensus_step` from a `StorageBid`. If we are performing a simple range proof, should this be treated as a constant?

Same applies to `secret_k`

Reply:
That needs to be discussed with @toghrulmaharramov . And we could address it in #47 since it's already taking care of the big refactor.

Is it ok?

Reply:
Yep, I'll take from this point with tog. Thx